.github/workflows/release.yml

```diff
@@ -1,98 +1,165 @@
-name: Release [WIP]
+name: Release 3 - Publish Release
+run-name: Publish Release ${{ inputs.version }}
 
 on:
-  push:
-    tags:
-    - 'v*'
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Metabase version (e.g. v0.46.3)'
+        type: string
+        required: true
+      commit:
+        description: 'A full-length commit SHA-1 hash'
+        required: true
 
 jobs:
-
-  trigger-docs-update:
-    runs-on: ubuntu-22.04
-    timeout-minutes: 10
-    steps:
-      - uses: actions/github-script@v6
-        with:
-          github-token: ${{ secrets.DOCS_UPDATE_TOKEN }}
-          script: |
-            github.rest.repos.createDispatchEvent({
-              owner: '${{ github.repository_owner }}',
-              repo: '${{ secrets.DOCS_REPO }}',
-              event_type: 'trigger-docs-update',
-              client_payload: {
-                version: '${{ github.ref_name }}'
-              }
-            });
-
   download-uberjar:
     runs-on: ubuntu-22.04
     timeout-minutes: 10
+    strategy:
+      matrix:
+        edition: [oss, ee]
     steps:
-    - name: Download Uberjar for ${{ github.ref_name }}
+    - name: Fail early on the incorrect version format
+      if: ${{ !(startsWith(inputs.version,'v0.') || startsWith(inputs.version,'v1.')) }}
       run: |
-        JAR_DOWNLOAD_URL=https://downloads.metabase.com/${{ github.ref_name }}/metabase.jar
-        if [[ ${{ github.ref_name }} == v1* ]]; then
-          JAR_DOWNLOAD_URL=https://downloads.metabase.com/enterprise/${{ github.ref_name }}/metabase.jar
-        fi
-        echo $JAR_DOWNLOAD_URL > url.txt
-        echo "----- Downloading Uberjar from $JAR_DOWNLOAD_URL -----"
-        curl -OL $JAR_DOWNLOAD_URL
-        stat ./metabase.jar
-        date | tee timestamp
+        echo "The version format is invalid!"
+        echo "It must start with either 'v0.' or 'v1.'."
+        echo "Please, try again."
+        exit 1
+    - name: find_release_artifact
+      id: find_release_artifact
+      uses: actions/github-script@v6
+      with:
+        result-encoding: string
+        script: |
+          const artifacts = await github.rest.actions.listArtifactsForRepo({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            name: `metabase-${{ matrix.edition }}-${{ inputs.commit }}-uberjar`,
```

Comment:
because we named the artifact with the commit, we can ensure that we get the right one with just the name